iOS5 Attempt/iTunes Search 2/iTunes Search 2/SearchResultsTableViewController.swift

```diff
@@ -0,0 +1,64 @@
+import UIKit
+
+class SearchResultsTableViewController: UITableViewController, UISearchBarDelegate {
+    
+    @IBOutlet weak var segmentedControl: UISegmentedControl!
+    
+    @IBOutlet weak var searchBar: UISearchBar!
+    
+    let searchResultsController = SearchResultController()
+    
+    let reuseIdentifier = "ResultCell"
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        searchBar.delegate = self
+    }
+    
+    // MARK: - Table view data source
+    
+    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return searchResultsController.searchResults.count
+    }
+    
+    
+    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = tableView.dequeueReusableCell(withIdentifier: reuseIdentifier, for: indexPath)
+        
+        let searchResult = searchResultsController.searchResults[indexPath.row]
+        
+        cell.textLabel?.text = searchResult.title
+        cell.detailTextLabel?.text = searchResult.creator
+        
+        guard let url = URL(string: searchResult.artwork),
+            let imageData = try? Data(contentsOf: url) else { return cell }
+        
+        cell.imageView?.image = UIImage(data: imageData)
```

Comment:
Getting data from a `URL` and converting it to a `UIImage` is taking time and freezing UI. You should move this operation to background queue and in the completion closure update `imageView` and reload table in main queue. This [article](https://www.raywenderlich.com/5370-grand-central-dispatch-tutorial-for-swift-4-part-1-2) could help.